### PR TITLE
Test discovery blockchain

### PR DESCRIPTION
**Pull request by hackaugusto**
Merge after: https://github.com/raiden-network/raiden/pull/664 [merged]

### CODE REVIEW

**Comment by LefterisJP** [comment]
Rebased on top of latest master.

**Comment by hackaugusto** [comment]
This is not working properly yet, some test are failing, adding the WIP tag.

**Comment by LefterisJP** [comment]
Thanks wanted to make sure that tests were failing also when it's rebased on master. Put a comment and also change the label when ready to have it reviewed.

**Comment by hackaugusto** [comment]
please review

File: raiden/tests/fixtures/blockchain.py [hunk 1]
```diff
@@ -148,8 +138,22 @@ def cached_genesis(request, blockchain_type):
...
+        ContractDiscovery(
+            chain.node_address,
+            chain.discovery(endpoint_discovery_address),
+        )
+        for chain in blockchain_services
+    ]
+
     raiden_apps = create_apps(
         blockchain_services,
+        endpoing_discovery_services,
```
Comment (konradkonrad) [inline raiden/tests/fixtures/blockchain.py]: `s/endpoing/endpoint/`

File: raiden/tests/fixtures/blockchain.py [hunk 2]
```diff
@@ -148,8 +138,22 @@ def cached_genesis(request, blockchain_type):
         register
     )
 
+    endpoint_discovery_address = deploy_service.deploy_contract(
+        'EndpointRegistry',
+        'EndpointRegistry.sol',
+    )
+
+    endpoing_discovery_services = [
```
Comment (konradkonrad) [inline raiden/tests/fixtures/blockchain.py]: `s/endpoing/endpoint/`

**Comment by hackaugusto** [comment]
Fixed the typos :)

**Comment by LefterisJP** [comment]
Rebased on latest master.

File: raiden/tests/fixtures/blockchain.py [hunk 3]
```diff
@@ -96,7 +86,7 @@ def dagpath():
     return os.path.expanduser('~/.ethash/full-R23-0000000000000000')
 
 
-@pytest.fixture(scope='session', autouse=True)
+@pytest.fixture(autouse=True)
```
Comment (LefterisJP) [inline raiden/tests/fixtures/blockchain.py]: Why did you remove the `session` scope here?

Comment (hackaugusto) [inline raiden/tests/fixtures/blockchain.py]: blockchain_type is not a session fixture, so pregenerate_dag shouldn't be one, I don't understand how that was not failing before. This change has no loss of behavior.

Comment (LefterisJP) [inline raiden/tests/fixtures/blockchain.py]: Ha ... I see. I assumed this pregenerates the DAG and as such we suddenly switched to pregenerating the DAG before each test with this change.

Reading the fixture code I see that is not the case. So I made  a mini PR to rename it: https://github.com/raiden-network/raiden/pull/705
